pkg/cableengine/cableengine.go

```diff
@@ -138,7 +138,13 @@ func (i *engine) InstallCable(endpoint types.SubmarinerEndpoint) error {
 		}
 
 		if util.GetClusterIDFromCableName(active) == endpoint.Spec.ClusterID {
-			return fmt.Errorf("found a pre-existing cable %q that belongs to this cluster %s", active, endpoint.Spec.ClusterID)
+			klog.V(log.DEBUG).Infof("Found a pre-existing cable %q that belongs to this cluster %s", active, endpoint.Spec.ClusterID)
+			preExistingEndpoint := endpoint
+			preExistingEndpoint.Spec.CableName = active
+			err = i.driver.DisconnectFromEndpoint(preExistingEndpoint)
+			if err != nil {
+				return err
+			}
```

Comment:
Shouldn't we obtain the actual previous endpoint info:

```suggestion
			activeEndpointInfo := i.getEndpointInfo(active, connections)
			if activeEndpointInfo != nil {
				err = i.driver.DisconnectFromEndpoint(types.SubmarinerEndpoint{*activeEndpointInfo})
				if err != nil {
					return err
				}
			}
```

Reply:
Also I think the `ClusterID` check should be an else to the preceding `if`. Otherwise if the `CableName` was the same but the `PublicIP` changed then it would fall thru to this code b/c the `ClusterID` would be the same. I don't think this is the intention. Previously it would've returned an error - with these changes it would try to disconnect again.  I think we should structure it like this:

```
for _, active := range activeConnections {
    ...
    activeEndpointInfo := i.getEndpointInfo(active, connections)
    if activeEndpointInfo == nil {
        klog.Warningf("No Connection found for active Endpoint cable name %q", active)
        continue
    }

    if active == endpoint.Spec.CableName {
        ...
    } else if util.GetClusterIDFromCableName(active) == endpoint.Spec.ClusterID {
        ...
    }
}
```

Reply:
Thats a good point @tpantelis, thanks.